### DISCUSSION TITLE
DeviceCheckManager for Apple App Attest verification

### DISCUSSION THREAD

**Discussion by PoloShock** [webauthn4j/webauthn4j#426] [General]
Reading the [reference](https://webauthn4j.github.io/webauthn4j/en/#apple-app-attest-attestation-verification), I can see there is a `deviceCheckManager` object used in the code sample, but I cannot see its way of initialization anywhere.

From reading the source code, I can see I can create it simply like this:
`DeviceCheckManager.createNonStrictDeviceCheckManager()`
Which I suppose is fine for development purposes, but is it a good idea to have this 'non strict manager', which internally uses `NullCertPathTrustworthinessValidator`, in production environment too? Can anyone please advice how I should initialize it for production environment? And does it need for example an [apple root certificate](https://www.apple.com/certificateauthority/private) as an input?

Thank you for any suggestions.

**Comment by ynojima** [comment]
Sorry for lack of documentation.
Yes, you need to initialize `DeviceCheckManager` with Apple App Attestation Root CA file in this way for production:
``` java
CertFileTrustAnchorsProvider certFileTrustAnchorsProvider = new CertFileTrustAnchorsProvider();
Path path = null; //TODO: set path to Apple App Attestation Root CA file
certFileTrustAnchorsProvider.setCertificates(Collections.singletonList(path));
deviceCheckManager = new DeviceCheckManager(new TrustAnchorCertPathTrustworthinessValidator(new TrustAnchorsResolverImpl(certFileTrustAnchorsProvider)));
```
